MapboxMobileEventsTests/MMEAPIClientTests_cedar.mm

```diff
@@ -0,0 +1,327 @@
+#import <Cedar/Cedar.h>
+#import <Foundation/Foundation.h>
+#import "MMEAPIClient.h"
+#import "MMEConstants.h"
+#import "MMEEvent.h"
+#import "MMECommonEventData.h"
+#import "MMETrustKitWrapper.h"
+#import "MMENSURLSessionWrapperFake.h"
+#import "MMEAPIClientFake.h"
+
+using namespace Cedar::Matchers;
+using namespace Cedar::Doubles;
+
+@interface MMEAPIClient (Tests)
+
+@property (nonatomic) id<MMENSURLSessionWrapper> sessionWrapper;
+@property (nonatomic, copy) NSURL *baseURL;
+@property (nonatomic) BOOL usesTestServer;
+@property (nonatomic) NSBundle *applicationBundle;
+@property (nonatomic, copy) NSString *userAgent;
+
+- (void)setupUserAgent;
+
+@end
+
+SPEC_BEGIN(MMEAPIClientSpec)
+
+describe(@"MMEAPIClient", ^{
+    
+    __block MMEAPIClient *apiClient;
+    
+    beforeEach(^{
+        apiClient = [[MMEAPIClient alloc] initWithAccessToken:@"access-token"
+                                                userAgentBase:@"user-agent-base"
+                                               hostSDKVersion:@"host-sdk-1"];
+    });
+    
+    describe(@"- testing API setup", ^{
+        context(@"when MMETelemetryTestServerURL is nil", ^{
+            
+            beforeEach(^{
+                spy_on([NSUserDefaults standardUserDefaults]);
+                [NSUserDefaults standardUserDefaults] stub_method(@selector(objectForKey:)).with(MMETelemetryTestServerURL).and_return(nil);
+                
+                apiClient = [[MMEAPIClient alloc] initWithAccessToken:@"access-token"
+                                                        userAgentBase:@"user-agent-base"
+                                                       hostSDKVersion:@"host-sdk-1"];
+                
+                apiClient.sessionWrapper should_not be_nil;
+            });
+            
+            it(@"should equal the BaseURL constant", ^{
+                apiClient.baseURL should equal([NSURL URLWithString:MMEAPIClientBaseURL]);
+            });
+            
+            afterEach(^{
```

Comment:
nit: I think I prefer to have the afterEach block just below the beforeEach block at the top of the context so that you can more easily read about setup and teardown in one place. Not a big deal though.